workers/unity/Packages/com.improbable.gdk.playerlifecycle/PlayerLifecycleHelper.cs

```diff
@@ -7,14 +8,44 @@ namespace Improbable.Gdk.PlayerLifecycle
     public static class PlayerLifecycleHelper
     {
         public static EntityBuilder AddPlayerLifecycleComponents(this EntityBuilder entityBuilder,
+            string workerId,
             string clientAccess,
             string serverAccess)
         {
             var clientHeartbeat = PlayerHeartbeatClient.Component.CreateSchemaComponentData();
             var serverHeartbeat = PlayerHeartbeatServer.Component.CreateSchemaComponentData();
+            var owningComponent = OwningWorker.Component.CreateSchemaComponentData(workerId);
+
             return entityBuilder
                 .AddComponent(clientHeartbeat, clientAccess)
-                .AddComponent(serverHeartbeat, serverAccess);
+                .AddComponent(serverHeartbeat, serverAccess)
+                .AddComponent(owningComponent, serverAccess);
+        }
+
+        public static bool IsOwningWorker(SpatialEntityId entityId, World workerWorld)
+        {
+            var entityManager = workerWorld.GetOrCreateManager<EntityManager>();
+            var worker = workerWorld.GetExistingManager<WorkerSystem>();
+
+            if (worker == null)
+            {
+                throw new InvalidOperationException("Provided World does not have an associated worker");
+            }
+
+            if (!worker.TryGetEntity(entityId.EntityId, out var entity))
+            {
+                throw new InvalidOperationException(
```

Comment:
Should this return false instead?

Reply:
I don't think so. This could only ever happen through gross misuse of the API

Reply:
Also we don't know that the answer is false. This is being asked for information you can't know. So saying no is strictly wrong.

Reply:
I see, I agree